packages/util-crypto/src/schnorrkel/deriveHard.ts

```diff
@@ -6,12 +6,13 @@ import { Keypair } from '../types';
 
 import '../polyfill';
 
-import { sr25519DeriveKeypairHard } from '@polkadot/wasm-crypto';
+import { sr25519DeriveKeypairHard, waitReady } from '@polkadot/wasm-crypto';
 
 import keypairFromU8a from './keypair/fromU8a';
 import keypairToU8a from './keypair/toU8a';
 
-export default function deriveHard (keypair: Keypair, chainCode: Uint8Array): Keypair {
+export default async function deriveHard (keypair: Keypair, chainCode: Uint8Array): Keypair {
+  await waitReady()
```

Comment:
The issue with the fix above is that the signature is changed from sync to async. So the return here is not `Promise<Keypair>`

If you use sr25519, it is up to the user to initialise the wasm correctly with a wait. (e.g. via the API, the isReady does that, outside of that you need to do it manually)